src/dos/drive_virtual.cpp

```diff
@@ -48,37 +49,28 @@ extern DOS_Shell *first_shell;
 
 class VFILE_Block;
 using vfile_block_t = std::shared_ptr<VFILE_Block>;
+using vfile_data_t  = std::shared_ptr<std::vector<uint8_t>>;
 
-class VFILE_Block {
+class VFILE_Block final {
 public:
 	std::string name = {};
 
-	uint8_t* data = nullptr;
+	vfile_data_t data = nullptr;
 
-	uint32_t size      = 0;
 	uint16_t date      = 0;
 	uint16_t time      = 0;
-	unsigned int onpos = 0;
+	unsigned int position = 0;
 
 	bool isdir = false;
 
 	vfile_block_t next = {};
 
-	~VFILE_Block();
+	~VFILE_Block() = default;
```

Comment:
This sure is nice to have these memory allocated vfiles cleaned up when the last reference goes out of scope!

Reply:
I needed a safe way to replace the Z:\AUTOEXEC.BAT content (when the code page changes) even if the file is open - so I used smart pointers for data. VFILE support indeed needs a rework - but now within this PR :)